pulsar-client-cpp/CMakeLists.txt

```diff
@@ -23,13 +23,20 @@ project (pulsar-cpp)
 option(LINK_STATIC "Link against static libraries" OFF)
 MESSAGE(STATUS "LINK_STATIC:  " ${LINK_STATIC})
 
+IF (CMAKE_BUILD_TYPE STREQUAL "")
```

Comment:
There's already a status message (at line 30) that prints the `CMAKE_BUILD_TYPE` variable. 

`RelWithDebInfo` is actually a release configuration with debug symbols: `-O3 -g` so I think it's appropriate as default. These are the same flags we have right now.